backend/imagekit/client/media.go

```diff
@@ -0,0 +1,345 @@
+		err = fmt.Errorf("http error %d: %v", response.StatusCode, response.Status)
+	}
+
+	return response, data, err
+}
+
+// Files retrieves media library files. Filter options can be supplied as FilesOrFolderParam.
+func (ik *ImageKit) Files(ctx context.Context, params FilesOrFolderParam, includeVersion bool) (*http.Response, *[]File, error) {
+	var SearchQuery string = `type = "file"`
+
+	if includeVersion {
+		SearchQuery = `type IN ["file", "file-version"]`
+	}
+	if params.SearchQuery != "" {
+		SearchQuery = params.SearchQuery
+	}
+
+	parameters := url.Values{}
+
+	parameters.Set("skip", fmt.Sprintf("%d", params.Skip))
+	parameters.Set("limit", fmt.Sprintf("%d", params.Limit))
+	parameters.Set("path", params.Path)
+	parameters.Set("searchQuery", SearchQuery)
+
+	data := &[]File{}
+
+	response, err := ik.HttpClient.CallJSON(ctx, &rest.Opts{
+		Method:     "GET",
+		Path:       "/files",
+		RootURL:    ik.Prefix,
+		Parameters: parameters,
+	}, nil, data)
+
+	if err != nil {
+		return response, data, err
+	}
+
+	if response.StatusCode != 200 {
+		err = ParseError(response)
+	}
+
+	return response, data, err
+}
+
+// DeleteFile removes file by FileId from media library
+func (ik *ImageKit) DeleteFile(ctx context.Context, fileId string) (*http.Response, error) {
+	var err error
+
+	if fileId == "" {
+		return nil, errors.New("fileId can not be empty")
+	}
+
+	response, err := ik.HttpClient.CallJSON(ctx, &rest.Opts{
+		Method:  "DELETE",
+		Path:    fmt.Sprintf("/files/%s", fileId),
+		RootURL: ik.Prefix,
+	}, nil, nil)
+
+	if err != nil {
+		return response, err
+	}
+
+	if response.StatusCode != 204 {
+		err = ParseError(response)
+	}
+
+	return response, err
+}
+
+// Folders retrieves media library files. Filter options can be supplied as FilesOrFolderParam.
+func (ik *ImageKit) Folders(ctx context.Context, params FilesOrFolderParam) (*http.Response, *[]Folder, error) {
+	var SearchQuery string = `type = "folder"`
+
+	if params.SearchQuery != "" {
+		SearchQuery = params.SearchQuery
+	}
+
+	parameters := url.Values{}
+
+	parameters.Set("skip", fmt.Sprintf("%d", params.Skip))
+	parameters.Set("limit", fmt.Sprintf("%d", params.Limit))
+	parameters.Set("path", params.Path)
+	parameters.Set("searchQuery", SearchQuery)
+
+	data := &[]Folder{}
+
+	resp, err := ik.HttpClient.CallJSON(ctx, &rest.Opts{
+		Method:     "GET",
+		Path:       "/files",
+		RootURL:    ik.Prefix,
+		Parameters: parameters,
+	}, nil, data)
+
+	if err != nil {
+		return resp, data, err
+	}
+
+	if resp.StatusCode != 200 {
+		err = ParseError(resp)
+	}
+
+	return resp, data, err
+}
+
+// CreateFolder creates a new folder in media library
+func (ik *ImageKit) CreateFolder(ctx context.Context, param CreateFolderParam) (*http.Response, error) {
+	var err error
+
+	if err = validator.Validate(&param); err != nil {
+		return nil, err
+	}
+
+	response, err := ik.HttpClient.CallJSON(ctx, &rest.Opts{
+		Method:  "POST",
+		Path:    "/folder",
+		RootURL: ik.Prefix,
+	}, param, nil)
```

Comment:
`nil` response so body needs closing or set `NoResponse`

Reply:
verified and updated